File: modules/mrlocallocal/src/main/java/com/soundcloud/android/mrlocallocal/data/MrLocalLocalResult.java
package com.soundcloud.android.mrlocallocal.data;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class MrLocalLocalResult {
    public abstract boolean wasSuccessful();
    public abstract boolean canBeRetried();
    public abstract String message();

    public static MrLocalLocalResult success() {
        return new AutoValue_MrLocalLocalResult(true, false, "🚀 MrLocalLocal verification successful!");
    }

    public static MrLocalLocalResult error(boolean canBeRetried, String message) {
        return new AutoValue_MrLocalLocalResult(false, canBeRetried, message);
    }

}

File: modules/mrlocallocal/src/main/java/com/soundcloud/android/mrlocallocal/data/MrLocalLocalResult.kt
package com.soundcloud.android.mrlocallocal.data

data class MrLocalLocalResult(
        val wasSuccessful: Boolean,
        val canBeRetried: Boolean,
        val message: String,
        val log: String
) {
    companion object {
        @JvmStatic
        fun success(): MrLocalLocalResult {
            return MrLocalLocalResult(true, false, "🚀 MrLocalLocal verification successful!", "")
        }

        @JvmStatic
        fun error(canBeRetried: Boolean, message: String, log: String): MrLocalLocalResult {
            return MrLocalLocalResult(false, canBeRetried, message, log)
        }
    }

    fun buildErrorMessage(): String {
        return """
            $message
            -------------
            $log
            """
    }
}
